src/debug.c

```diff
@@ -2223,7 +2227,10 @@ static void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
         /* If this thread already owns the lock (meaning we crashed during handling a signal)
          * log that the crash report can't be generated. */
         serverLogRawFromHandler(LL_WARNING,
-            "Crashed running signal handler. Can't continue to generate the crash report");
+            "Crashed running signal handler. Providing reduced version of crash report.");
+#ifdef HAVE_BACKTRACE
+        logStackTrace(NULL, 1);
+#endif
```

Comment:
maybe a reduced version is not just the stack trace?
we still wanna print the signal number, si_code, registers, etc?
maybe a better solution is to remove the `return`, and set a flag to skip some parts of the report?

also, i think one of the concerns leading to the creation of this block that stops a recursive signal handler was an issue with the stack trace collection mechanism itself. maybe when this happens, we wanna pass a flag to logStackTrace to only print the stack trace of the current thread.
i.e. it's simpler, there's less chance for it to fail again, and it's certainly sufficient for an investigation of a recursive signal

Reply:
Yeah, I read through the past discussion and wasn't entirely clear why this was just exiting without doing more work. I can print some more stuff.

Not sure about the details around logStackTrace, but printing out just the current thread when the deadlock fired seems doable.

Reply:
**On Linux systems** - Be aware that if a crash occurs during the collection of stack traces, the call to `writeStacktraces` -> `ThreadsManager_runOnThreads` will return NULL. This is because the state flag is set to `IN_PROGRESS`.
If you decide to print only the current thread's stack trace, please note that the implementation for this functionality is currently only available for operating systems other than Linux.
Additionally, the logStackTrace function opens the log file. is it safe to do it when log file is already open?

Reply:
so let's make the necessary changes to print a crash lot that excludes INFO, and includes only a stack trace of the current thread.